pkg/bridgeclient/app.go

```diff
@@ -224,69 +206,115 @@ func getMessagesFromCloud(serverURL, clientID string) ([]v1.BridgeMessage, error
 	return msglist, nil
 }
 
-func sendMessageToCloud(msg *nats.Msg, serverURL string, clientID string, ceEnabled bool) {
-	log.Debugf("Sending Msg NB %s", msg.Subject)
-
-	msgFormat := msgs.GetMsgFormat()
-	status, err := msgFormat.ValidateMsgFormat(msg.Data, ceEnabled)
+func subscribeToOutboundMessages(serverURL, clientID string) (*nats.Subscription, error) {
+	nc := bridgemodel.GetNatsConnection()
+	subj := fmt.Sprintf("%s.>", msgs.NATSSYNC_MESSAGE_PREFIX)
+	sub, err := nc.SubscribeSync(subj)
 	if err != nil {
-		log.Errorf("Error validating the cloud event message: %s", err.Error())
-		return
+		return nil, err
+	}
+	go handleOutboundMessages(sub, serverURL, clientID)
+	return sub, nil
+}
+
+// handleOutboundMessages  This pulls messages off the queue and groups a bunch of them to push them together
+// if we have to wait more than N ms for a message, we will go ahead and send what we have
+// or if we get more than N messages, we will send them along
+func handleOutboundMessages(subscription *nats.Subscription, serverURL, clientID string) {
+	timeoutStr := pkg.GetEnvWithDefaults("NATSSYNC_MSG_WAIT_TIMEOUT", "5")
+	maxMsgHoldStr := pkg.GetEnvWithDefaults("NATSSYNC__MAX_MSG_HOLD", "512")
+	waitTimeout, numErr := strconv.ParseInt(timeoutStr, 10, 16)
+	if numErr != nil {
+		waitTimeout = 5
 	}
-	if !status {
-		log.Errorf("Cloud event message validation failed, ignoring the message...")
-		return
+	maxQueueSize, numErr := strconv.ParseInt(maxMsgHoldStr, 10, 16)
+	if numErr != nil {
+		waitTimeout = 512
 	}
 
-	url := fmt.Sprintf("%s/bridge-server/1/message-queue/%s", serverURL, clientID)
-	natmsg := bridgemodel.NatsMessage{Reply: msg.Reply, Subject: msg.Subject, Data: msg.Data}
-	envelope, enverr := msgs.PutObjectInEnvelope(natmsg, clientID, pkg.CLOUD_ID)
-	if enverr != nil {
-		log.Errorf("Error putting msg in envelope %s", enverr.Error())
-		return
+	msgList := make([]*nats.Msg, 0)
+	keepGoing := true
+	for keepGoing {
```

Comment:
Do we need some kind of a break condition?

Reply:
yep.... oops